pkgs/misc/cups/libppd.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, cups
+, libcupsfilters
+, ghostscript
+, pkg-config
+, poppler_utils
+, mupdf
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libppd";
+  version = "2.0b2";
+
+  src = fetchFromGitHub {
+    owner = "OpenPrinting";
+    repo = "libppd";
+    rev = version;
+    hash = "sha256-4KRJfzK/xeWlXmhUOlvGHsMmKqF6R4e5BiIA+h2UTII=";
+  };
+
+  configureFlags = [
+    "--with-mutool-path=${lib.getBin mupdf}/bin/mutool"
+    "--with-pdftops=pdftops"
+    "--with-pdftops-path=${lib.getBin poppler_utils}/bin/pdftops"
+    "--with-gs-path=${lib.getBin ghostscript}/bin/gs"
+    "--with-pdftocairo-path=${lib.getBin poppler_utils}/bin/pdftocairo"
+    "--with-ippfind-path=${lib.getBin cups}/bin/ippfind"
+    "--disable-acroread"
+    "--localstatedir=/var"
+    "--sysconfdir=/etc"
+  ];
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [
+    cups
+    ghostscript
+    poppler_utils
+    mupdf
+    libcupsfilters
+    zlib
+  ];
+
+  # Fix missing includes on darwin
+  postPatch = lib.strings.optionalString (stdenv.hostPlatform.libc != "glibc")
+    ''
+      substituteInPlace ppd/ppd-collection.cxx \
+        --replace \
+        '#include <sys/wait.h>' \
+        "#include <sys/wait.h>''\n#include <signal.h>''\n#include <libgen.h>''\nextern char **environ;"
+
+      substituteInPlace Makefile.am \
+        --replace 'libppd_la_LIBADD = ' 'libppd_la_LIBADD = ''$(ZLIB_LIBS) ' \
+        --replace 'libppd_la_CFLAGS = ' 'libppd_la_CFLAGS = ''$(ZLIB_CFLAGS) '
+
+      substituteInPlace configure.ac \
+        --replace \
+        'PKG_CHECK_MODULES([LIBCUPSFILTERS], [libcupsfilters])' \
+        'PKG_CHECK_MODULES([LIBCUPSFILTERS], [libcupsfilters])''\nPKG_CHECK_MODULES([ZLIB], [zlib])'
```

Comment:
I wouldn't mind if those where patches like said earlier already.

Reply:
I've converted them to patches now :slightly_smiling_face: